gen_vehicle_model.py

```diff
@@ -85,17 +103,36 @@
 
     # yaml_out = open(args.yaml_file, "w", encoding="utf-8")
 
+    ext_attributes_list = args.extended_attributes.split(",")
+    if len(ext_attributes_list) > 0:
+        vspec.model.vsstree.VSSNode.whitelisted_extended_attributes = ext_attributes_list
+        print(
+            f"Known extended attributes: {', '.join(ext_attributes_list)}")
+
     try:
         print("Loading vspec...")
         tree = vspec.load_tree(
             args.vspec_file,
             include_dirs,
             merge_private=False,
-            break_on_unknown_attribute=True,
+            break_on_unknown_attribute=strict,
             break_on_name_style_violation=strict,
             expand_inst=False,
         )
 
+        for overlay in args.overlays:
+            print(f"Applying VSS overlay from {overlay}...")
+            overlay_tree = vspec.load_tree(
+                overlay,
+                include_dirs,
+                merge_private=False,
+                break_on_unknown_attribute=strict,
+                break_on_name_style_violation=strict,
+                expand_inst=False
+            )
+            vspec.merge_tree(tree, overlay_tree)
```

Comment:
If you want to use the merged tree for model generation i think you missed to reassign the tree from line 114

Reply:
No, `merge_tree` does not return something. It merges the overlay into the base.